flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/sources/tsextractors/ExistingField.java

```diff
@@ -42,7 +42,12 @@
 /**
  * Converts an existing {@link Long}, {@link java.sql.Timestamp}, or timestamp formatted
  * java.lang.String field (e.g., "2018-05-28 12:34:56.000") into a rowtime attribute.
+ *
+ * @deprecated This class will not be supported in the new source design around {@link
+ *     org.apache.flink.table.connector.source.DynamicTableSource}. See FLIP-95 for more
+ *     information.
  */
+@Deprecated
 @PublicEvolving
 public final class ExistingField extends TimestampExtractor {
```

Comment:
It is used at [1] which turns out that StreamSQLTestProgram.GeneratorTableSource should be replaced with the new design. Does it make sense to do it within this PR?

[1] https://github.com/apache/flink/blob/d6967dd7301e82fa102f756e16635dabce1c550d/flink-end-to-end-tests/flink-stream-sql-test/src/main/java/org/apache/flink/sql/tests/StreamSQLTestProgram.java#L236

Reply:
Hi @JingGe, thanks for your reminder. 

> Does it make sense to do it within this PR?

I think we can open a dedicated task to migrate the E2E test to the new design since except for `ExistingField`, `StreamSQLTestProgram` also refers to some already deprecated APIs like `StreamTableSource`, `DefinedRowtimeAttributes`, `DefinedFieldMapping`, and `TableSchema` etc.

WDYT?

Reply:
works for me.